doc/run_settings.rst

```diff
@@ -0,0 +1,159 @@
+************
+Run Settings
+************
+
+========
+Overview
+========
+SmartSim allows you to configure the run settings of a job (``Model`` or ``Ensemble``)
+through a ``RunSettings`` object. Each SmartSim `launcher` interfaces with a number of
+``RunSettings`` objects. For the local `launcher`, SmartSim supports
+a base ``RunSettings`` object. For the WLM `launchers`, SmartSim supports a variety of
+``RunSettings`` child classes.
+
+* Navigate to :ref:`local section<Local>` to configure run settings locally
+* Navigate to :ref:`wlm section<HPC>` to configure run settings for a HPC
+
+A run settings object is initialized through the ``Experiment.create_run_settings()`` function.
+This function accepts a `run_command` argument: the command to run the executable.
+If `run_command` is set to `auto`, SmartSim will attempt to match a run command on the
+system with a ``RunSettings`` class. If found, the class corresponding to
+that `run_command` will be created and returned.
+If the `run_command` is passed a recognized run command (ex. 'Slurm') the ``RunSettings``
+instance will be a child class such as ``SrunSettings``.
+If not supported by smartsim, the base ``RunSettings`` class will be
+created and returned with the specified `run_command` and `run_args` evaluated literally.
+You may also specify `mpi_run` or `mpi_exec` to the `run_command` argument. This will return
+the associated child class.
+
+Once a ``RunSettings`` object is initialized, you have access to helper functions that
+further configure the run settings of a job. The following chart shows helper functions
+associated with the ``SrunSettings`` object:
+
+.. list-table:: SrunSettings Helper Functions
+   :widths: 25 55 25
+   :header-rows: 1
+
+   * - ``SrunSettings`` func
```

Comment:
I notice that we can use CSV files as the source of table information in the sphinx guide: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/tables.html#csv-files

I wonder if we might benefit by generating code documentation and transforming to csv. That could ensure that the arguments are always up to date and documentation is updated automagically

Reply:
So instead I am going to lead the user to the list of funcs in the SmartSimAPI instead of having a table here. I think it might make things less lengthy